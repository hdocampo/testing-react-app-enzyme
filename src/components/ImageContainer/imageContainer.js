import React from 'react';

const ImageContainer = ({ src, alt, className, onClick }) => (
    <div>
        <img src={src}
            className={className}
            alt={alt}
            onClick={onClick}
        />
    </div>
)

export default ImageContainer;