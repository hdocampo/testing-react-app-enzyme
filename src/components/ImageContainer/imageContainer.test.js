import React from 'react';
import { mount, shallow, render } from 'enzyme';

import ImageContainer from './imageContainer';

const clickFn = jest.fn();

describe('ImageContainer', () => {
    it('should render the ImageContainer component', () => {
        const component = shallow(<ImageContainer onClick={clickFn} />);

        component.find('img')
            .simulate('click');

        expect(clickFn).toHaveBeenCalled();

        expect(component).toMatchSnapshot();

        component.unmount();
    })
})