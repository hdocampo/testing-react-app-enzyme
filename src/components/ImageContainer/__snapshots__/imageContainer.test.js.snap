// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageContainer should render the ImageContainer component 1`] = `
<div>
  <img
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />
</div>
`;
